AmplifyPlugins/Core/AWSPluginsCoreTests/Model/Decorator/AuthRuleDecorator/ModelWithOwnerAuthAndMultiGroup.swift

```diff
@@ -0,0 +1,242 @@
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost {
+          onCreateOIDCMultiGroupPost {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        XCTAssertTrue(document.variables.isEmpty)
+    }
+
+    func testOnCreateSubscription_InDifferentAppWithModeratorsAndEditors() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000",
+                      "https://differentapp.com/claims/groups": ["Moderators", "Editors"]] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost {
+          onCreateOIDCMultiGroupPost {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        XCTAssertTrue(document.variables.isEmpty)
+    }
+
+    func testOnCreateSubscription_InDifferentAppWithAdminsFromMyApp() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000",
+                      "https://myapp.com/claims/groups": ["Admins"]] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost {
+          onCreateOIDCMultiGroupPost {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        XCTAssertTrue(document.variables.isEmpty)
```

Comment:
```suggestion
        XCTAssertTrue(document.variables.isEmpty, "variables should be empty since group claims match auth rule")
```

Reply:
Changed to:
"variables should be empty since claim group value matches the auth rule schema"